### PR TITLE
Issue #21: Add context-specific escaper helpers

### PR DESCRIPTION
**Pull request by robopuff**

### CODE REVIEW

File: src/Extension/EscaperExtension.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+     * - escapeCss($string) : string
+     * - escapeUrl($string) : string
+     *
+     * @param Engine $engine
+     * @return void
+     * @throws InvalidArgumentException
+     */
+    public function register(Engine $engine)
+    {
+        $escaper = new Escaper();
```
Comment (weierophinney) [inline src/Extension/EscaperExtension.php]: We should likely allow providing an encoding to use with the escaper. While the default of UTF-8 is fine, being able to specify another is something we definitely want within a template engine.

That could be done via a constructor parameter.

**Review by weierophinney — CHANGES_REQUESTED** [review]
I've noted that we need to be able to specify the encoding as an optional feature. That would likely also require a factory for the extension, that would pull the encoding from configuration. That factory should be registered in the `ConfigProvider`.

Finally, this does not register the extension by default. I'm thinking it should, as an added feature of this bridge. As such, you'd need to update the `PlatesEngineFactory` to do so. Alternately, document that the extension exists, and demonstrate how to register it (via the `plates.extensions` configuration array, which allows specifying service names of extensions).

File: src/Extension/EscaperExtension.php (hunk 1 above)
Comment (robopuff) [reply src/Extension/EscaperExtension.php]: @weierophinney  You think that user should be able to provide a encoding in plates views as well?
For example: `escapeHtml($string, $encoding=null)`

**Review by geerteltink — APPROVED** [review]

**Review by weierophinney — APPROVED** [review]
The only other change I would make is to register the escaper extension in the `ConfigProvider` by default; I can do that on merge, though.

Nice work!

**Comment by weierophinney** [comment]
Thanks, @robopuff; merged to develop for release with 1.4.0.

**Pull request closed without merging**